main.go

```diff
@@ -164,21 +164,50 @@ func dumpJSONFile(file string, pretty bool, schema bool) error {
 	return nil
 }
 
-func getFilesContentHash(files []string) (string, error) {
+func getFilesContentHash(files []string, tmp string) (string, error) {
+	sha1 := sha1.New()
+	buf := make([]byte, 4*sha1.BlockSize())
 
-	var contents []byte
-	for _, f := range files {
-		c, err := os.ReadFile(f)
-		if err != nil {
-			return "", err
-		}
-		contents = append(contents, c...)
-	}
+	for i := 0; i < len(files); i++ {
+		if files[i] == "" {
+			i += 1
+			b, err := io.ReadAll(os.Stdin)
```

Comment:
I'm still confused why you're doing temp buffers here.

You should be able to hook up os.Stdin directly to sha1.Writer without a temp buffer.

Reply:
I mentioned io.Copy before, here's another example of that https://stackoverflow.com/a/45123990/1507139

Reply:
Which is pulled from the sha1 docs https://pkg.go.dev/crypto/sha256#New

Reply:
Although this isn't a file since os.Stdin is a writer I think it will work the same way

Reply:
btw just one

Reply:
Oh I  just didn't see that comment prior to the first commit